### ISSUE TITLE
Using twisted.cred.credentials results in deprecation warnings in Twisted 2.4

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1748]
|<img alt="szumo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| szumo reported|
|-|-|
|Trac ID|trac#1748|
|Type|defect|
|Created|2006-05-25 11:20:53Z|

twisted.cred.credentials imports and uses deprecated twisted.python.components.Interface instead of zope.interface.Interface.

This results in annoying warnings under Twisted 2.4:

```
>>> import twisted.cred.credentials
>>> class Test(twisted.cred.credentials.ICredentials):
...   pass
...
__main__:1: ComponentsDeprecationWarning: twisted.python.components.Interface is deprecated in Twisted 2.3. Use zope.interface.Interface instead. (also note that you should not use 'self' as the first arg on function decls in z.i.Interfaces).
```

(This is actually a special case of [#11150](https://github.com/twisted/twisted/issues/11150))

Attachments:

* [twisted-credentials.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ddb/ddb6158990bb9a24621a9e1785be80d8bf4ae482/d8dc11c6aee53b81f3f0027f025ad7a086a6cdb9.patch) (504 bytes) - added by zaheerm on 2006-06-22 14:34:59Z - patch to fix this issue

<details><summary>Searchable metadata</summary>

```
trac-id__1748 1748
type__defect defect
reporter__szumo szumo
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__cred_interfaces cred,interfaces
time__1148556053000000 1148556053000000
changetime__1151771305000000 1151771305000000
version__2_0 2.0
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight set status to `closed`|
|-|-|

It's actually a special case of [#11082](https://github.com/twisted/twisted/issues/11082), but is also fixed by [#11150](https://github.com/twisted/twisted/issues/11150).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
